pxr/usd/bin/usdtree/usdtree.py

```diff
@@ -0,0 +1,228 @@
+    
+    numAttrs = len(attrs)
+    for i, attr in enumerate(attrs):
+        if i < numAttrs - 1:
+            _Msg('{}{} :--{}'.format(prefix, attrStep, attr))
+        else:
+            _Msg('{}{} `--{}'.format(prefix, attrStep, attr))
+
+
+def PrintChildren(args, prim, iterType, prefix):
+    children = _GetChildren(prim, iterType)
+    numChildren = len(children)
+    for i, child in enumerate(children):
+        if i < numChildren - 1:
+            PrintPrim(args, child, iterType, prefix, isLast=False)
+            PrintChildren(args, child, iterType, prefix + ' |  ')
+        else:
+            PrintPrim(args, child, iterType, prefix, isLast=True)
+            PrintChildren(args, child, iterType, prefix + '    ')
+
+
+def PrintStage(args, stage):
+    _Msg('USD')
+    PrintChildren(args, stage.GetPseudoRoot(), 'UsdPrim', '')
+
+
+def PrintLayer(args, layer):
+    _Msg('USD')
+    PrintChildren(args, layer.pseudoRoot, 'SdfPrimSpec', '')
+
+
+def PrintTree(args, path):
+    if args.flatten:
+        from pxr import Usd
+        popMask = (None if args.populationMask is None else Usd.StagePopulationMask())
+        if popMask:
+            for mask in args.populationMask:
+                popMask.Add(mask)
+        if popMask:
+            if args.unloaded:
+                stage = Usd.Stage.OpenMasked(path, popMask, Usd.Stage.LoadNone)
+            else:
+                stage = Usd.Stage.OpenMasked(path, popMask)
+        else:
+            if args.unloaded:
+                stage = Usd.Stage.Open(path, Usd.Stage.LoadNone)
+            else:
+                stage = Usd.Stage.Open(path)
+        if args.flattenLayerStack:
+            from pxr import UsdUtils
+            stage = UsdUtils.FlattenLayerStack(stage)
+        PrintStage(args, stage)
+    else:
+        from pxr import Sdf
+        layer = Sdf.Layer.FindOrOpen(path)
+        PrintLayer(args, layer)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Writes the tree structure of a USD file. The default is to inspect a single USD file. '
+        'Use the --flatten argument to see the flattened (or composed) Stage tree.')
+
+    parser.add_argument('inputPath',
+        help='The input file path to usdtree')
+    parser.add_argument(
+        '--unloaded', action='store_true',
+        dest='unloaded',
+        help='Do not load payloads')
+    parser.add_argument(
+        '--attributes', '-a', action='store_true',
+        dest='attributes',
+        help='Display authored attributes')
+    parser.add_argument(
+        '--metadata', '-m', action='store_true',
+        dest='metadata',
+        help='Display authored metadata')
+    parser.add_argument(
+        '--types', '-t', action='store_true',
+        dest='types',
+        help='Display prim types')
+    parser.add_argument(
+        '-f', '--flatten', action='store_true', help='Compose stages with the '
+        'input files as root layers and write their flattened content')
```

Comment:
inputFile is a singular input, so we should talk about 'stage' and 'layer' above, not 'stages' and 'layers'